crates/nu-cmd-lang/src/core_commands/return_.rs

```diff
@@ -40,17 +40,11 @@ impl Command for Return {
         _input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
         let return_value: Option<Value> = call.opt(engine_state, stack, 0)?;
-        if let Some(value) = return_value {
-            Err(ShellError::Return {
-                span: call.head,
-                value: Box::new(value),
-            })
-        } else {
-            Err(ShellError::Return {
-                span: call.head,
-                value: Box::new(Value::nothing(call.head)),
-            })
-        }
+        let value = return_value.unwrap_or(Value::nothing(call.head));
```

Comment:
Not sure if it makes much of a difference in release mode

```suggestion
        let value = return_value.unwrap_or_else(|| Value::nothing(call.head));
```

Reply:
I would ***assume*** it doesn't, since `Value::nothing()` doesn't do any meaningful work (e.g., no allocation). It  can even be a `const` function if we wanted it to.

Reply:
True

Making it `const` would be a `nu-protocol` stability promise, not sure if you would actually gain much right now from.